code/src/components/GameScreen.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+
+import assets from '../assets/compass.png'
+import { nextAction } from '../reducers/gameactions'
+
+import Summary from './Summary'
+
+const GameScreen = () => {
+  const gameStart = useSelector(store => store.gameactions.gamestart)
+  const loading = useSelector(store => store.gameactions.loading)
+
+  const dispatch = useDispatch()
+
+  return (
+    <div className='gamescreen-container'>
+      {loading &&
+        <span className='image-container'>
+          <img src={assets} alt='compass' />
+        </span>
+      }
+      {!loading && (
+        <div className='description-container'>
+          <p className='fade-in'>{gameStart.description}</p>
+          {gameStart.actions.map((action) => (
+            <div key={action.description}>
+              <p className='fade-in'>{action.description}</p>
+              <button
+                className='fade-in'
+                onClick={() => dispatch(nextAction(action.direction))}
+              >
+                Go {action.direction}
+              </button>
+            </div>
+          ))}
+          {gameStart.coordinates === '1,3' && <Summary />}
```

Comment:
Smart idea to render the summary based on the coordinates.

Reply:
It is a nice solution, but it is not a future-proof one. What if the map structure will change on the backend (the game will grow for example)? However, in this case, we can safely assume that we are in the final room if there are **no remaining actions**

So our logic could be something like:

```jsx
if (actions.length === 0) {
   return <GameOver />;
}else{
...
```
or 
```jsx
...
actions.length === 0 && <EndScreen/>
...
```